packages/manager/apps/catalog-revamp/src/utils/utils.ts

```diff
@@ -0,0 +1,140 @@
+export type Product = {
+  id: number;
+  name: string;
+  description: string;
+  lang: string;
+  categories: string[];
+  url: string;
+  regionTags: string[];
+  productName: string;
+  order: string;
+  universe: string;
+  category: string;
+};
+
+type ProductKey = 'universe' | 'category' | 'name';
+
+export const groupItemsByUniverse = (itemsToGroup: Product[]) => {
+  return itemsToGroup.reduce<Record<string, Product[]>>((acc, item) => {
+    if (!acc[item.universe]) acc[item.universe] = [];
+    acc[item.universe].push(item);
+    return acc;
+  }, {});
+};
+
+export const filterByEnum = (
+  product: Product,
+  query: string[],
+  property: ProductKey,
+) => {
+  return query.length === 0 || query.includes(product[property]);
+};
```

Comment:
```suggestion
export const filterByProperty = (
  product: Product,
  query: string[],
  property: ProductKey,
) => query.length === 0 || query.includes(product[property]);
```

Reply:
This is not an enum
Use the right terms so we can read the code easily

Reply:
Done